openapi/api_auth.template.yaml

```diff
@@ -1,102 +1,812 @@
-      - spid_mobile_phone
-  CieUserIdentity:
-    type: object
-    title: CIE User Identity
-    description: Describes the user's profile while it's authenticated with CIE.
-    properties:
-      family_name:
-        type: string
-      fiscal_code:
-        $ref: '#/definitions/FiscalCode'
-      name:
-        type: string
-      date_of_birth:
-        type: string
-        format: date
-    required:
-      - family_name
-      - fiscal_code
-      - name
-      - date_of_birth
-  UserIdentity:
-    x-one-of: true
-    allOf:
-      - $ref: "#/definitions/SpidUserIdentity"
-      - $ref: "#/definitions/CieUserIdentity"
-consumes:
-  - application/json
-produces:
-  - application/json
-securityDefinitions:
-  Bearer:
-    type: apiKey
-    name: Authorization
-    in: header
+            $ref: "#/components/schemas/LollipopMethod"
+        - name: x-pagopa-lollipop-original-url
+          in: header
+          description: The url of the endpoint called by IO app
+          required: true
+          schema:
+            $ref: "#/components/schemas/LollipopOriginalURL"
+        - name: content-digest
+          in: header
+          description: The body digest, if any, as defined in https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-digest-headers-10
+          required: true
+          schema:
+            $ref: "#/components/schemas/LollipopContentDigest"
+        - name: signature-input
+          in: header
+          description: The signature input, needed to verify the `signature` header
+          required: true
+          schema:
+            $ref: "#/components/schemas/LollipopSignatureInput"
+        - name: signature
+          in: header
+          description: The signature of the HTTP request, signed by the client with its private key.
+          required: true
+          schema:
+            $ref: "#/components/schemas/LollipopSignature"
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/SignMessagePayload"
+      responses:
+        "200":
+          description: "Success response"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/SignMessageResponse"
+        "400":
+          description: Bad request
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ProblemJson"
+        "401":
+          description: Token null or invalid.
+        "500":
+          description: There was an error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ProblemJson"
+
+components:
+  schemas:
+    # Definitions from the digital citizenship APIs
+    AcceptedTosVersion:
+      $ref: "https://raw.githubusercontent.com/pagopa/io-functions-commons/openapi-v3-definitions/openapi/definitions_v3.yaml#/AcceptedTosVersion"
```

Comment:
Since we are referencing a remote spec, do we still need the template file and the generation step?

Reply:
In order to generate a clean openapi spec that will be loaded into the api group declaration into the APIM, is better have a generated spec without remote references.